ui/app/routes/vault/cluster/secrets/backend/list.js

```diff
@@ -19,12 +19,16 @@ export default Ember.Route.extend({
   templateName: 'vault/cluster/secrets/backend/list',
 
   beforeModel() {
-    const { backend } = this.paramsFor('vault.cluster.secrets.backend');
-    const backendModel = this.store.peekRecord('secret-engine', backend);
-    const type = backendModel && backendModel.get('type');
+    let { backend } = this.paramsFor('vault.cluster.secrets.backend');
+    let { secret } = this.paramsFor(this.routeName);
+    let backendModel = this.store.peekRecord('secret-engine', backend);
+    let type = backendModel && backendModel.get('type');
     if (!type || !SUPPORTED_BACKENDS.includes(type)) {
       return this.transitionTo('vault.cluster.secrets');
     }
+    if (this.routeName === 'vault.cluster.secrets.backend.list' && !secret.endsWith('/')) {
```

Comment:
Why is the `this.routeName` check needed? Are you subclassing this route elsewhere, or is there something else I don't understand here?

Reply:
Yep, you pretty much hit the nail on the head 🔨 - https://github.com/hashicorp/vault/blob/df18871704fe869e9be45b542a6b1eb2fe46c293/ui/app/router.js#L74-L82 at the root the secret is "", but you can also navigate down the tree, and you can't have the same route for both cases, so we have `-root` versions of the secret CRUD routes and `list-root.js` is this file re-exported.